llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.cpp

```diff
@@ -198,11 +200,7 @@ uint32_t getModuleDeviceLibReqMask(const Module &M) {
   // Device libraries will be enabled only for spir-v module.
   if (!llvm::Triple(M.getTargetTriple()).isSPIR())
     return 0;
-  // 0x1 means sycl runtime will link and load libsycl-fallback-assert.spv as
-  // default. In fact, default link assert spv is not necessary but dramatic
-  // perf regression is observed if we don't link any device library. The perf
-  // regression is caused by a clang issue.
-  uint32_t ReqMask = 0x1;
+  uint32_t ReqMask = 0;
```

Comment:
I'm just curious, what was the performance issue described in the old comment?  Is that fixed now?

Reply:
Hi, @gmlueck 
Long time ago, clang introduced one bug and would lead to a fp related flag missing in sycl device code and would lead to perf drop for sycl programs. However, libsycl-fallback-cassert.spv did include this flag and when linked with user's device code, it would save the perf by happy coincidence. This bug has nothing to do with libdevice and has been fixed.

Thanks very much.